.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,95 @@
+# For most projects, this workflow file will not need changing; you simply need
```

Comment:
Just a note (doesn't need to be implemented now) but we should add license checks to this repo and the SPDX headers.

Reply:
We have a `lint` target run during `test.yaml` workflow, just need to add the needed `addlicense` call to it to check